includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -167,6 +182,15 @@ public function get_settings( $latest = false ) {
 		return $settings;
 	}
 
+	public function save_settings() {
+		if ( $this->storing_settings_enabled() && empty( $this->order_settings ) && ! empty( $this->settings ) && ! empty( $this->order ) ) {
```

Comment:
This won't work when regenerating a document, as the order settings are not empty in this case. Therefore the new settings are not saved.

Reply:
Because you want to regenerate the document, so you need to use (and save) the latest settings.